### PR TITLE
[WIP] Opgops 1935

### PR DESCRIPTION
**Pull request by brettminnie**
0x644 Images

This does not exist

### CODE REVIEW

File: base0x644/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# Log aggregation
+ADD  docker/sources.list /etc/apt/sources.list
+RUN gpg --keyserver hkp://ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.2/gosu-$(dpkg --print-architecture)" \
+  && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.2/gosu-$(dpkg --print-architecture).asc" \
+  && gpg --verify /usr/local/bin/gosu.asc \
+  && rm /usr/local/bin/gosu.asc \
+  && chmod +x /usr/local/bin/gosu
+
+RUN  rm -f /etc/apt/sources.list.d/* && \
```
Comment (tim-mcarthur-moj) [inline base0x644/Dockerfile]: we should switch to using apt

File: base0x644/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     unzip zip && \
+     apt-get clean && apt-get autoremove && \
+     rm -rf /var/lib/cache/* /var/lib/log/* /tmp/* /var/tmp/*
+
+# Pip installs
+# Install awscli to override the system one which is broken (with python 3.4.3 as of Sept 2015)
+RUN pip install -U pip && pip install awscli Beaver==36.2.0
+
+# ConfD install
+RUN  wget -O /usr/bin/confd https://github.com/kelseyhightower/confd/releases/download/v0.9.0/confd-0.9.0-linux-amd64 && \
```
Comment (tim-mcarthur-moj) [inline base0x644/Dockerfile]: update to use latest stable 0.11 version

File: base0x644/docker/beaver.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[beaver]
+redis_url: redis://monitoring:6379/0
+redis_namespace: logstash
+logstash_version: 1
+queue_timeout: 3600
+
+[/var/log/ebs-snapshot.log]
+format: json
+type: ebs-snapshot
+tags: ebs-snapshot, master
```
Comment (tim-mcarthur-moj) [inline base0x644/docker/beaver.conf]: why is the master tag in here? should this not be changed to include the stack tag as per the other log file configs for beaver

File: nginx0x644/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
+FROM registry.service.opg.digital/opguk/base0x644
+
+RUN  apt-get install -y libpcre3-dev zlib1g-dev libssl-dev libossp-uuid-dev uuid && \
```
Comment (tim-mcarthur-moj) [inline nginx0x644/Dockerfile]: change to using apt command

File: nginx0x644/Makefile [hunk 5]
```diff
@@ -0,0 +1,21 @@
+.PHONY: build test
+
+registryUrl ?= registry.service.opg.digital
+oldRegistryUrl ?= registry.service.dsd.io
```
Comment (tim-mcarthur-moj) [inline nginx0x644/Makefile]: can be removed

File: nginx0x644/Makefile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+image = opguk/nginx0x644
+
+build:
+	docker build $(no-cache) -t "$(registryUrl)/$(image)" .
+	docker tag "$(registryUrl)/$(image)" "$(registryUrl)/$(image):$(newtag)"
```
Comment (tim-mcarthur-moj) [inline nginx0x644/Makefile:9]: remove all refs to old registry

File: nginx0x644/docker/nginx/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,98 @@
+# For more information on configuration, see:
+#   * Official English Documentation: http://nginx.org/en/docs/
+#   * Official Russian Documentation: http://nginx.org/ru/docs/
+
+worker_processes  1;
```
Comment (tim-mcarthur-moj) [inline nginx0x644/docker/nginx/nginx.conf]: this should be templated

File: nginx0x644/docker/service/nginx/run [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+mkdir -p /var/log/nginx
+
+mkdir -p /var/log/app
```
Comment (tim-mcarthur-moj) [inline nginx0x644/docker/service/nginx/run]: this is done by base, so isn't needed

File: php-fpm0x644/docker/confd/templates/pool.app.conf.tmpl [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+
+; The number of child processes to be created when pm is set to 'static' and the
+; maximum number of child processes to be created when pm is set to 'dynamic'.
+; This value sets the limit on the number of simultaneous requests that will be
+; served. Equivalent to the ApacheMaxClients directive with mpm_prefork.
+; Equivalent to the PHP_FCGI_CHILDREN environment variable in the original PHP
+; CGI.
+; Note: Used when pm is set to either 'static' or 'dynamic'
+; Note: This value is mandatory.
+pm.max_children = {{getv "/opg/php/pool/children/max"}}
```
Comment (tim-mcarthur-moj) [inline php-fpm0x644/docker/confd/templates/pool.app.conf.tmpl]: this should have a sane default

**Review by tim-mcarthur-moj — CHANGES_REQUESTED** [review]

File: base0x644/Dockerfile (hunk 1 above)
Comment (brettminnie) [reply base0x644/Dockerfile]: apt raises a cli warning at present

File: base0x644/docker/beaver.conf (hunk 3 above)
Comment (brettminnie) [reply base0x644/docker/beaver.conf]: Direct import for now, I am going to review

File: nginx0x644/Makefile (hunk 6 above)
Comment (brettminnie) [reply nginx0x644/Makefile:9]: See OPGOPS-1937

File: nginx0x644/docker/nginx/nginx.conf (hunk 7 above)
Comment (brettminnie) [reply nginx0x644/docker/nginx/nginx.conf]: Most of these should be, I'll review today

**Review by tim-mcarthur-moj — APPROVED** [review]

**Comment by brettminnie** [comment]
Closing this as we're moving to the separate repo

**Pull request closed without merging**